File: 2002/3/9-3.py
#!/usr/bin/env python

import sys

for line in sys.stdin:
    groups = set()
    count = 0
    for l in line:
        if l == 'A':
            count += 1
        else:
            if count > 0:
                groups.add(count)
            count = 0
    if count > 0:
        groups.add(count)
    print len(groups)

File: 2002/3/9-3.rb
#!/usr/bin/env ruby

STDIN.each do |line|
  groups = []
  count = 0
  line.chars do |c|
    if c == 'A'
      count += 1
    else
      groups << count if count > 0
      count = 0
    end
  end
  groups << count if count > 0
  puts groups.uniq.length
end
